### ISSUE TITLE
[5.6.y/arm64] - rpi4 /sys/class/thermal/thermal_zone0/temp not working.

### ISSUE THREAD

**Issue by asavah** [raspberrypi/linux#3524]
Just upgraded with a freshly built kernel from rpi-5.6.y branch, arm64.
No userspace updates were made, just the kernel.
Noticed 2 bugs, will report the second one in a separate issue.

Hardware:  Machine model: Raspberry Pi 4 Model B Rev 1.1 , 4GB
OS: own pet project, arm64

Bug: cat: /sys/class/thermal/thermal_zone0/temp: Invalid argument

It seems to be impossible to read cpu temp on rpi4 with 5.6.y kernel , 5.5 was OK
```
cat /sys/class/thermal/thermal_zone0/temp
cat: /sys/class/thermal/thermal_zone0/temp: Invalid argument
```

dmesg:
```
    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd083]
[    0.000000] Linux version 5.6.0-k2 (asavah@kross-1910) (gcc version 9.3.1 20200323 (GCC)) #1 SMP Wed Apr 1 15:51:45 EEST 2020
[    0.000000] Machine model: Raspberry Pi 4 Model B Rev 1.1
[    0.000000] cma: Reserved 64 MiB at 0x000000002ac00000
[    0.000000] NUMA: No NUMA configuration found
[    0.000000] NUMA: Faking a node at [mem 0x0000000000000000-0x00000000fbffffff]
[    0.000000] NUMA: NODE_DATA [mem 0xfb863100-0xfb864fff]
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000000000-0x000000003fffffff]
[    0.000000]   DMA32    [mem 0x0000000040000000-0x00000000fbffffff]
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x000000002fffffff]
[    0.000000]   node   0: [mem 0x0000000040000000-0x00000000fbffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x00000000fbffffff]
[    0.000000] On node 0 totalpages: 966656
[    0.000000]   DMA zone: 3072 pages used for memmap
[    0.000000]   DMA zone: 0 pages reserved
[    0.000000]   DMA zone: 196608 pages, LIFO batch:63
[    0.000000]   DMA32 zone: 12032 pages used for memmap
[    0.000000]   DMA32 zone: 770048 pages, LIFO batch:63
[    0.000000] percpu: Embedded 31 pages/cpu s86552 r8192 d32232 u126976
[    0.000000] pcpu-alloc: s86552 r8192 d32232 u126976 alloc=31*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3
[    0.000000] Detected PIPT I-cache on CPU0
[    0.000000] CPU features: kernel page table isolation disabled by kernel configuration
[    0.000000] spectrev2 mitigation disabled by kernel configuration
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 951552
[    0.000000] Policy zone: DMA32
[    0.000000] Kernel command line: coherent_pool=1M 8250.nr_uarts=1 cma=64M bcm2708_fb.fbwidth=1920 bcm2708_fb.fbheight=1280 bcm2708_fb.fbswap=1 smsc95xx.macaddr=DC:A6:32:0A:92:54 vc_mem.mem_base=0x3ec00000 vc_mem.mem_size=0x40000000  dwc_otg.lpm_enable=0 console=tty1 root=/dev/mmcblk0p2 rootfstype=ext4 fsck.repair=yes rootwait quiet
[    0.000000] Dentry cache hash table entries: 524288 (order: 10, 4194304 bytes, linear)
[    0.000000] Inode-cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] software IO TLB: mapped [mem 0x26c00000-0x2ac00000] (64MB)
[    0.000000] Memory: 3638576K/3866624K available (9982K kernel code, 1164K rwdata, 3728K rodata, 960K init, 661K bss, 162512K reserved, 65536K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] ftrace: allocating 31809 entries in 125 pages
[    0.000000] ftrace: allocated 125 pages with 6 groups
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=8 to nr_cpu_ids=4.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 100 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] GIC: Using split EOI/Deactivate mode
[    0.000000] random: get_random_bytes called from start_kernel+0x308/0x498 with crng_init=0
[    0.000000] arch_timer: cp15 timer(s) running at 54.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0xc743ce346, max_idle_ns: 440795203123 ns
[    0.000005] sched_clock: 56 bits at 54MHz, resolution 18ns, wraps every 4398046511102ns
[    0.000217] Console: colour dummy device 80x25
[    0.000242] printk: console [tty1] enabled
[    0.000323] Calibrating delay loop (skipped), value calculated using timer frequency.. 108.00 BogoMIPS (lpj=54000)
[    0.000338] pid_max: default: 32768 minimum: 301
[    0.000529] Mount-cache hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.000583] Mountpoint-cache hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.001705] Disabling memory control group subsystem
[    0.003150] rcu: Hierarchical SRCU implementation.
[    0.004137] smp: Bringing up secondary CPUs ...
[    0.004857] Detected PIPT I-cache on CPU1
[    0.004913] CPU1: Booted secondary processor 0x0000000001 [0x410fd083]
[    0.005792] Detected PIPT I-cache on CPU2
[    0.005831] CPU2: Booted secondary processor 0x0000000002 [0x410fd083]
[    0.006615] Detected PIPT I-cache on CPU3
[    0.006652] CPU3: Booted secondary processor 0x0000000003 [0x410fd083]
[    0.006763] smp: Brought up 1 node, 4 CPUs
[    0.006772] SMP: Total of 4 processors activated.
[    0.006781] CPU features: detected: 32-bit EL0 Support
[    0.006789] CPU features: detected: CRC32 instructions
[    0.045533] CPU: All CPU(s) started at EL2
[    0.045679] alternatives: patching kernel code
[    0.047432] devtmpfs: initialized
[    0.054753] Enabled cp15_barrier support
[    0.054778] Enabled setend support
[    0.054933] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 1911260446275000 ns
[    0.054956] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.056758] pinctrl core: initialized pinctrl subsystem
[    0.057301] thermal_sys: Registered thermal governor 'step_wise'
[    0.057869] NET: Registered protocol family 16
[    0.060201] DMA: preallocated 1024 KiB pool for atomic allocations
[    0.060906] cpuidle: using governor menu
[    0.061164] hw-breakpoint: found 6 breakpoint and 4 watchpoint registers.
[    0.061296] ASID allocator initialised with 65536 entries
[    0.061372] Serial: AMBA PL011 UART driver
[    0.063787] bcm2835-mbox fe00b880.mailbox: mailbox enabled
[    0.071555] raspberrypi-firmware soc:firmware: Attached to firmware from 2020-03-10 15:58, variant start_x
[    0.072564] raspberrypi-firmware soc:firmware: Firmware hash is 74d7f53b091e465285f38fb96896a8fc328e4c47
[    0.097123] HugeTLB registered 1.00 GiB page size, pre-allocated 0 pages
[    0.097138] HugeTLB registered 32.0 MiB page size, pre-allocated 0 pages
[    0.097147] HugeTLB registered 2.00 MiB page size, pre-allocated 0 pages
[    0.097155] HugeTLB registered 64.0 KiB page size, pre-allocated 0 pages
[    0.102519] bcm2835-dma fe007000.dma: DMA legacy API manager, dmachans=0x1
[    0.104760] vgaarb: loaded
[    0.105102] SCSI subsystem initialized
[    0.105254] usbcore: registered new interface driver usbfs
[    0.105291] usbcore: registered new interface driver hub
[    0.105351] usbcore: registered new device driver usb
[    0.105411] pps_core: LinuxPPS API ver. 1 registered
[    0.105418] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.106744] clocksource: Switched to clocksource arch_sys_counter
[    0.989959] VFS: Disk quotas dquot_6.6.0
[    0.990041] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.990155] FS-Cache: Loaded
[    0.990526] CacheFiles: Loaded
[    0.997870] NET: Registered protocol family 2
[    0.998508] tcp_listen_portaddr_hash hash table entries: 2048 (order: 3, 32768 bytes, linear)
[    0.998549] TCP established hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    0.998753] TCP bind hash table entries: 32768 (order: 7, 524288 bytes, linear)
[    0.999170] TCP: Hash tables configured (established 32768 bind 32768)
[    0.999438] UDP hash table entries: 2048 (order: 4, 65536 bytes, linear)
[    0.999492] UDP-Lite hash table entries: 2048 (order: 4, 65536 bytes, linear)
[    0.999764] NET: Registered protocol family 1
[    1.000468] RPC: Registered named UNIX socket transport module.
[    1.000476] RPC: Registered udp transport module.
[    1.000482] RPC: Registered tcp transport module.
[    1.000489] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    1.000506] PCI: CLS 0 bytes, default 64
[    1.002282] hw perfevents: enabled with armv8_cortex_a72 PMU driver, 7 counters available
[    1.003743] Initialise system trusted keyrings
[    1.003994] workingset: timestamp_bits=44 max_order=20 bucket_order=0
[    1.011191] FS-Cache: Netfs 'nfs' registered for caching
[    1.011992] NFS: Registering the id_resolver key type
[    1.012020] Key type id_resolver registered
[    1.012027] Key type id_legacy registered
[    1.012042] nfs4filelayout_init: NFSv4 File Layout Driver Registering...
[    1.013114] Key type asymmetric registered
[    1.013124] Asymmetric key parser 'x509' registered
[    1.013159] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 249)
[    1.013362] io scheduler mq-deadline registered
[    1.013370] io scheduler kyber registered
[    1.013488] io scheduler bfq registered
[    1.016384] brcm-pcie fd500000.pcie: host bridge /scb/pcie@7d500000 ranges:
[    1.016407] brcm-pcie fd500000.pcie:   No bus range found for /scb/pcie@7d500000, using [bus 00-ff]
[    1.016456] brcm-pcie fd500000.pcie:      MEM 0x0600000000..0x0603ffffff -> 0x00f8000000
[    1.016508] brcm-pcie fd500000.pcie:   IB MEM 0x0000000000..0x00bfffffff -> 0x0000000000
[    1.060827] brcm-pcie fd500000.pcie: link up, 5 GT/s x1 (SSC)
[    1.061108] brcm-pcie fd500000.pcie: PCI host bridge to bus 0000:00
[    1.061122] pci_bus 0000:00: root bus resource [bus 00-ff]
[    1.061136] pci_bus 0000:00: root bus resource [mem 0x600000000-0x603ffffff] (bus address [0xf8000000-0xfbffffff])
[    1.061181] pci 0000:00:00.0: [14e4:2711] type 01 class 0x060400
[    1.061367] pci 0000:00:00.0: PME# supported from D0 D3hot
[    1.064565] pci 0000:00:00.0: bridge configuration invalid ([bus 00-00]), reconfiguring
[    1.064819] pci 0000:01:00.0: [1106:3483] type 00 class 0x0c0330
[    1.064943] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x00000fff 64bit]
[    1.065302] pci 0000:01:00.0: PME# supported from D0 D3cold
[    1.070715] pci_bus 0000:01: busn_res: [bus 01-ff] end is updated to 01
[    1.070770] pci 0000:00:00.0: BAR 8: assigned [mem 0x600000000-0x6000fffff]
[    1.070787] pci 0000:01:00.0: BAR 0: assigned [mem 0x600000000-0x600000fff 64bit]
[    1.070815] pci 0000:00:00.0: PCI bridge to [bus 01]
[    1.070830] pci 0000:00:00.0:   bridge window [mem 0x600000000-0x6000fffff]
[    1.070890] pci 0000:00:00.0: enabling device (0000 -> 0002)
[    1.070911] pci 0000:01:00.0: enabling device (0000 -> 0002)
[    1.071350] bcm2708_fb soc:fb: FB found 1 display(s)
[    1.116710] Console: switching to colour frame buffer device 240x80
[    1.130910] bcm2708_fb soc:fb: Registered framebuffer for display 0, size 1920x1280
[    1.133129] Serial: 8250/16550 driver, 1 ports, IRQ sharing enabled
[    1.134579] iproc-rng200 fe104000.rng: hwrng registered
[    1.134782] vc-mem: phys_addr:0x00000000 mem_base=0x3ec00000 mem_size:0x40000000(1024 MiB)
[    1.135392] gpiomem-bcm2835 fe200000.gpiomem: Initialised: Registers at 0xfe200000
[    1.135683] cacheinfo: Unable to detect cache hierarchy for CPU 0
[    1.136571] Loading iSCSI transport class v2.0-870.
[    1.138273] libphy: Fixed MDIO Bus: probed
[    1.138699] bcmgenet fd580000.ethernet: failed to get enet clock
[    1.138713] bcmgenet fd580000.ethernet: GENET 5.0 EPHY: 0x0000
[    1.138754] bcmgenet fd580000.ethernet: failed to get enet-wol clock
[    1.138768] bcmgenet fd580000.ethernet: failed to get enet-eee clock
[    1.149767] libphy: bcmgenet MII bus: probed
[    1.193843] unimac-mdio unimac-mdio.-19: Broadcom UniMAC MDIO bus
[    1.195003] xhci_hcd 0000:01:00.0: xHCI Host Controller
[    1.195029] xhci_hcd 0000:01:00.0: new USB bus registered, assigned bus number 1
[    1.195800] xhci_hcd 0000:01:00.0: hcc params 0x002841eb hci version 0x100 quirks 0x0000001000000890
[    1.196516] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.06
[    1.196528] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    1.196538] usb usb1: Product: xHCI Host Controller
[    1.196548] usb usb1: Manufacturer: Linux 5.6.0-k2 xhci-hcd
[    1.196557] usb usb1: SerialNumber: 0000:01:00.0
[    1.197047] hub 1-0:1.0: USB hub found
[    1.197135] hub 1-0:1.0: 1 port detected
[    1.197579] xhci_hcd 0000:01:00.0: xHCI Host Controller
[    1.197596] xhci_hcd 0000:01:00.0: new USB bus registered, assigned bus number 2
[    1.197613] xhci_hcd 0000:01:00.0: Host supports USB 3.0 SuperSpeed
[    1.198133] usb usb2: New USB device found, idVendor=1d6b, idProduct=0003, bcdDevice= 5.06
[    1.198144] usb usb2: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    1.198153] usb usb2: Product: xHCI Host Controller
[    1.198162] usb usb2: Manufacturer: Linux 5.6.0-k2 xhci-hcd
[    1.198171] usb usb2: SerialNumber: 0000:01:00.0
[    1.198604] hub 2-0:1.0: USB hub found
[    1.198664] hub 2-0:1.0: 4 ports detected
[    1.199793] dwc_otg: version 3.00a 10-AUG-2012 (platform bus)
[    1.199971] dwc_otg: FIQ enabled
[    1.199978] dwc_otg: NAK holdoff enabled
[    1.199984] dwc_otg: FIQ split-transaction FSM enabled
[    1.200002] Module dwc_common_port init
[    1.200245] usbcore: registered new interface driver uas
[    1.200317] usbcore: registered new interface driver usb-storage
[    1.200460] mousedev: PS/2 mouse device common for all mice
[    1.200501] i2c /dev entries driver
[    1.201063] usbcore: registered new interface driver i2c-tiny-usb
[    1.201585] bcm2835-wdt bcm2835-wdt: Broadcom BCM2835 watchdog timer
[    1.203252] sdhci: Secure Digital Host Controller Interface driver
[    1.203260] sdhci: Copyright(c) Pierre Ossman
[    1.203780] mmc-bcm2835 fe300000.mmcnr: could not get clk, deferring probe
[    1.204138] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.208391] ledtrig-cpu: registered to indicate activity on CPUs
[    1.208435] hid: raw HID events driver (C) Jiri Kosina
[    1.208493] usbcore: registered new interface driver usbhid
[    1.208496] usbhid: USB HID core driver
[    1.208609] Initializing XFRM netlink socket
[    1.208890] NET: Registered protocol family 10
[    1.209461] Segment Routing with IPv6
[    1.209499] NET: Registered protocol family 17
[    1.209616] Key type dns_resolver registered
[    1.209885] registered taskstats version 1
[    1.209892] Loading compiled-in X.509 certificates
[    1.212967] uart-pl011 fe201000.serial: there is not valid maps for state default
[    1.213036] uart-pl011 fe201000.serial: cts_event_workaround enabled
[    1.213087] fe201000.serial: ttyAMA0 at MMIO 0xfe201000 (irq = 13, base_baud = 0) is a PL011 rev2
[    1.216899] bcm2835-power bcm2835-power: Broadcom BCM2835 power domains driver
[    1.217699] rtc-ds1307: probe of 1-0068 failed with error -121
[    1.218018] mmc-bcm2835 fe300000.mmcnr: mmc_debug:0 mmc_debug2:0
[    1.218022] mmc-bcm2835 fe300000.mmcnr: DMA channel allocated
[    1.263147] mmc1: queuing unknown CIS tuple 0x80 (2 bytes)
[    1.264658] mmc1: queuing unknown CIS tuple 0x80 (3 bytes)
[    1.266172] mmc1: queuing unknown CIS tuple 0x80 (3 bytes)
[    1.268919] mmc1: queuing unknown CIS tuple 0x80 (7 bytes)
[    1.270426] mmc1: queuing unknown CIS tuple 0x80 (3 bytes)
[    1.281427] mmc0: SDHCI controller on fe340000.emmc2 [fe340000.emmc2] using ADMA
[    1.282223] hctosys: unable to open rtc device (rtc0)
[    1.282227] of_cfs_init
[    1.282285] of_cfs_init: OK
[    1.283590] Waiting for root device /dev/mmcblk0p2...
[    1.308826] random: fast init done
[    1.333438] mmc1: new high speed SDIO card at address 0001
[    1.389555] mmc0: new ultra high speed DDR50 SDHC card at address aaaa
[    1.390352] mmcblk0: mmc0:aaaa SC16G 14.8 GiB
[    1.393633]  mmcblk0: p1 p2 p3
[    1.401693] EXT4-fs (mmcblk0p2): mounted filesystem with ordered data mode. Opts: (null)
[    1.401770] VFS: Mounted root (ext4 filesystem) readonly on device 179:2.
[    1.402328] devtmpfs: mounted
[    1.403679] Freeing unused kernel memory: 960K
[    1.403861] Run /sbin/init as init process
[    1.403864]   with arguments:
[    1.403867]     /sbin/init
[    1.403869]   with environment:
[    1.403872]     HOME=/
[    1.403874]     TERM=linux
[    1.516792] usb 1-1: new high-speed USB device number 2 using xhci_hcd
[    1.531863] usb 1-1: New USB device found, idVendor=2109, idProduct=3431, bcdDevice= 4.21
[    1.531878] usb 1-1: New USB device strings: Mfr=0, Product=1, SerialNumber=0
[    1.531889] usb 1-1: Product: USB2.0 Hub
[    1.533392] hub 1-1:1.0: USB hub found
[    1.533753] hub 1-1:1.0: 4 ports detected
[    1.808181] systemd[1]: System time before build time, advancing clock.
[    1.828880] systemd[1]: systemd 245-177-g12ef8fb+ running in system mode. (+PAM -AUDIT -SELINUX -IMA -APPARMOR -SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=unified)
[    1.829423] systemd[1]: Detected architecture arm64.
[    1.834576] systemd[1]: Set hostname to <rpi4>.
[    1.836332] systemd[1]: Hardware watchdog 'Broadcom BCM2835 Watchdog timer', version 0
[    1.836349] systemd[1]: Set hardware watchdog to 14s.
[    2.422592] systemd[1]: Queued start job for default target Kross System Setup target.
[    2.423644] random: systemd: uninitialized urandom read (16 bytes read)
[    2.425066] systemd[1]: Created slice system-getty.slice.
[    2.425285] random: systemd: uninitialized urandom read (16 bytes read)
[    2.425871] systemd[1]: Created slice system-modprobe.slice.
[    2.425925] random: systemd: uninitialized urandom read (16 bytes read)
[    2.426403] systemd[1]: Created slice system-wg\x2dquick.slice.
[    2.426925] systemd[1]: Created slice User and Session Slice.
[    2.427216] systemd[1]: Started Dispatch Password Requests to Console Directory Watch.
[    2.427444] systemd[1]: Started Forward Password Requests to Wall Directory Watch.
[    2.427649] systemd[1]: Reached target Local Encrypted Volumes.
[    2.427812] systemd[1]: Reached target NFS client services.
[    2.427911] systemd[1]: Reached target Paths.
[    2.428006] systemd[1]: Reached target Slices.
[    2.428095] systemd[1]: Reached target Swap.
[    2.435360] systemd[1]: Listening on Process Core Dump Socket.
[    2.435658] systemd[1]: Listening on initctl Compatibility Named Pipe.
[    2.476177] systemd[1]: Condition check resulted in Journal Audit Socket being skipped.
[    2.476925] systemd[1]: Listening on Journal Socket (/dev/log).
[    2.477447] systemd[1]: Listening on Journal Socket.
[    2.478122] systemd[1]: Listening on Network Service Netlink Socket.
[    2.478559] systemd[1]: Listening on udev Control Socket.
[    2.478942] systemd[1]: Listening on udev Kernel Socket.
[    2.479334] systemd[1]: Listening on User Database Manager Socket.
[    2.482171] systemd[1]: Mounting Huge Pages File System...
[    2.485126] systemd[1]: Mounting POSIX Message Queue File System...
[    2.487937] systemd[1]: Mounting Kernel Debug File System...
[    2.490719] systemd[1]: Mounting Kernel Trace File System...
[    2.495260] systemd[1]: Mounting Temporary Directory (/tmp)...
[    2.497505] systemd[1]: Starting Create list of static device nodes for the current kernel...
[    2.499816] systemd[1]: Starting Load Kernel Module drm...
[    2.503056] systemd[1]: Starting Journal Service...
[    2.506197] systemd[1]: Condition check resulted in Load Kernel Modules being skipped.
[    2.506503] systemd[1]: Condition check resulted in FUSE Control File System being skipped.
[    2.508357] systemd[1]: Mounting Kernel Configuration File System...
[    2.510701] systemd[1]: Starting Remount Root and Kernel File Systems...
[    2.513431] systemd[1]: Starting Apply Kernel Variables...
[    2.516137] systemd[1]: Starting udev Coldplug all Devices...
[    2.521290] systemd[1]: Mounted Huge Pages File System.
[    2.521705] systemd[1]: Mounted POSIX Message Queue File System.
[    2.522104] systemd[1]: Mounted Kernel Debug File System.
[    2.522518] systemd[1]: Mounted Kernel Trace File System.
[    2.522935] systemd[1]: Mounted Temporary Directory (/tmp).
[    2.523845] systemd[1]: Finished Create list of static device nodes for the current kernel.
[    2.524337] systemd[1]: Mounted Kernel Configuration File System.
[    2.543208] systemd[1]: modprobe@drm.service: Succeeded.
[    2.543640] systemd[1]: Finished Load Kernel Module drm.
[    2.544778] EXT4-fs (mmcblk0p2): re-mounted. Opts: (null)
[    2.549324] systemd[1]: Finished Remount Root and Kernel File Systems.
[    2.565076] systemd[1]: Starting Rebuild Hardware Database...
[    2.565186] systemd[1]: Condition check resulted in Platform Persistent Storage Archival being skipped.
[    2.568026] systemd[1]: Starting Load/Save Random Seed...
[    2.570202] systemd[1]: Starting Create System Users...
[    2.610667] systemd[1]: Finished Apply Kernel Variables.
[    2.611684] systemd[1]: Started Journal Service.
[    2.643068] systemd-journald[110]: Received client request to flush runtime journal.
[    2.647299] systemd-journald[110]: File /var/log/journal/707ff15bf8844b2ca4514f5687cf0cf1/system.journal corrupted or uncleanly shut down, renaming and replacing.
[    3.967755] random: crng init done
[    3.967764] random: 7 urandom warning(s) missed due to ratelimiting
[    4.785810] vchiq: module is from the staging directory, the quality is unknown, you have been warned.
[    4.797791] Registered IR keymap rc-rc6-mce
[    4.805424] IR RC6 protocol handler initialized
[    4.828386] rc rc0: gpio_ir_recv as /devices/platform/ir-receiver@12/rc/rc0
[    4.828480] rc rc0: lirc_dev: driver gpio_ir_recv registered at minor = 0, raw IR receiver, no transmitter
[    4.828561] input: gpio_ir_recv as /devices/platform/ir-receiver@12/rc/rc0/input0
[    4.909343] vc_sm_cma: module is from the staging directory, the quality is unknown, you have been warned.
[    4.910338] bcm2835_vc_sm_cma_probe: Videocore shared memory driver
[    4.910358] [vc_sm_connected_init]: start
[    4.915629] [vc_sm_connected_init]: installed successfully
[    4.920773] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[    4.932042] mc: Linux media interface: v0.10
[    4.972588] videodev: Linux video capture interface: v2.00
[    5.039308] cfg80211: Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[    5.061803] platform regulatory.0: Direct firmware loaded: regulatory.db
[    5.068195] bcm2835_mmal_vchiq: module is from the staging directory, the quality is unknown, you have been warned.
[    5.068197] bcm2835_mmal_vchiq: module is from the staging directory, the quality is unknown, you have been warned.
[    5.069812] platform regulatory.0: Direct firmware loaded: regulatory.db.p7s
[    5.083871] EXT4-fs (mmcblk0p3): mounted filesystem with ordered data mode. Opts: (null)
[    5.085103] brcmfmac: brcmf_fw_alloc_request: using brcm/brcmfmac43455-sdio for chip BCM4345/6
[    5.085450] usbcore: registered new interface driver brcmfmac
[    5.099465] bcm2835_codec: module is from the staging directory, the quality is unknown, you have been warned.
[    5.102044] bcm2835_v4l2: module is from the staging directory, the quality is unknown, you have been warned.
[    5.116679] brcmfmac mmc1:0001:1: Direct firmware loaded: brcm/brcmfmac43455-sdio.bin
[    5.121306] brcmfmac mmc1:0001:1: Direct firmware loaded: brcm/brcmfmac43455-sdio.raspberrypi,4-model-b.txt
[    5.267712] brcmfmac: brcmf_fw_alloc_request: using brcm/brcmfmac43455-sdio for chip BCM4345/6
[    5.270293] brcmfmac mmc1:0001:1: Direct firmware loaded: brcm/brcmfmac43455-sdio.clm_blob
[    5.281416] brcmfmac: brcmf_c_preinit_dcmds: Firmware: BCM4345/6 wl0: Nov 13 2019 04:39:05 version 7.45.198 (r723338 CY) FWID 01-369a6798
[    5.579136] bpfilter: Loaded bpfilter_umh pid 259
[    5.581224] Started bpfilter
[    5.581233] testing the buffer
[    5.581255] testing the buffer
[    5.582003] testing the buffer
[    5.610548] testing the buffer
[    5.729042] bcm2835-codec bcm2835-codec: Device registered as /dev/video10
[    5.729077] bcm2835-codec bcm2835-codec: Loaded V4L2 decode
[    5.732276] bcm2835-codec bcm2835-codec: Device registered as /dev/video11
[    5.732307] bcm2835-codec bcm2835-codec: Loaded V4L2 encode
[    5.738143] bcm2835-codec bcm2835-codec: Device registered as /dev/video12
[    5.738173] bcm2835-codec bcm2835-codec: Loaded V4L2 isp
[    5.942218] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    5.993231] 8021q: 802.1Q VLAN Support v1.8
[    6.002879] bcmgenet fd580000.ethernet: configuring instance for external RGMII (RX delay)
[    6.005053] bcmgenet fd580000.ethernet eth0: Link is Down
[    6.033895] br37: port 1(vlan37) entered blocking state
[    6.033903] br37: port 1(vlan37) entered disabled state
[    6.034078] device vlan37 entered promiscuous mode
[    6.034082] device eth0 entered promiscuous mode
[    6.370352] br37: port 2(wlan0) entered blocking state
[    6.370361] br37: port 2(wlan0) entered disabled state
[    6.370520] device wlan0 entered promiscuous mode
[    6.828399] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
[    6.828500] br37: port 2(wlan0) entered blocking state
[    6.828505] br37: port 2(wlan0) entered forwarding state
[    6.828639] IPv6: ADDRCONF(NETDEV_CHANGE): br37: link becomes ready
[    7.350749] rc rc0: two consecutive events of type space
[   11.105951] bcmgenet fd580000.ethernet eth0: Link is Up - 1Gbps/Full - flow control off
[   11.105991] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   11.106412] IPv6: ADDRCONF(NETDEV_CHANGE): vlan20: link becomes ready
[   11.106619] IPv6: ADDRCONF(NETDEV_CHANGE): vlan37: link becomes ready
[   11.106715] br37: port 1(vlan37) entered blocking state
[   11.106764] br37: port 1(vlan37) entered forwarding state
[   15.528935] wireguard: WireGuard 1.0.0 loaded. See www.wireguard.com for information.
[   15.528945] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
```

firmware:
```
vcgencmd version
Mar 10 2020 15:58:31
Copyright (c) 2012 Broadcom
version 74d7f53b091e465285f38fb96896a8fc328e4c47 (clean) (release) (start_x)
```

**Comment by asavah** [comment]
Figured this one out.
Upstream added a separate driver bcm2711_thermal.
This driver need to be enabled, however for this driver even to be selectable in menuconfig MFD_SYSCON needs to be enabled first.

Currently arm64 bcm2711_defconfig does not enable said drivers but it does enable unneeded bcm2835_thermal (irrelevant for the issue, but still it's an unneded driver being built).

Enabling these 2 options
```
CONFIG_BCM2711_THERMAL=y
CONFIG_MFD_SYSCON=y
```
make `cat /sys/class/thermal/thermal_zone0/temp` great again.

Feel free to close once you read this )

**Comment by pelwell** [comment]
Thanks for the report and fix - see aad3538.
